misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/comparator/ClassComparator.java

```diff
@@ -0,0 +1,296 @@
+
+        Node newDocs = newMeta.flatMap(MetadataNode::documentationString).orElse(null);
+        Node oldDocs = oldMeta.flatMap(MetadataNode::documentationString).orElse(null);
+        DocumentationComparator documentationComparator = new DocumentationComparator(newDocs, oldDocs);
+        documentationComparator.computeDiff().ifPresent(metadataDiffs::add);
+
+        NodeList<AnnotationNode> newAnnots = newMeta.map(MetadataNode::annotations).orElse(null);
+        NodeList<AnnotationNode> oldAnnots = oldMeta.map(MetadataNode::annotations).orElse(null);
+        DumbNodeListComparator<AnnotationNode> annotsComparator = new DumbNodeListComparator<>(newAnnots, oldAnnots,
+                DiffKind.SERVICE_ANNOTATION.toString());
+        annotsComparator.computeDiff().ifPresent(metadataDiffs::add);
+
+        return metadataDiffs;
+    }
+
+    /**
+     * Analyzes and returns the diff for changes on `public`, `isolated`, `distinct`, `readonly`, `client` and
+     * `service` qualifiers.
+     */
+    private List<Diff> compareClassQualifiers() {
+        List<Diff> qualifierDiffs = new ArrayList<>();
+
+        // analyzes public qualifier changes
+        Optional<Token> newPublicQual = newNode.visibilityQualifier();
+        Optional<Token> oldPublicQual = oldNode.visibilityQualifier();
+        if (newPublicQual.isPresent() && oldPublicQual.isEmpty()) {
+            NodeDiffBuilder qualifierDiffBuilder = new NodeDiffImpl.Builder<Node>(newPublicQual.get(), null);
+            qualifierDiffBuilder
+                    .withVersionImpact(SemverImpact.MINOR)
+                    .withMessage("'public' qualifier is added to class '" + getClassName() + "'")
+                    .build()
+                    .ifPresent(qualifierDiffs::add);
+        } else if (newPublicQual.isEmpty() && oldPublicQual.isPresent()) {
+            NodeDiffBuilder qualifierDiffBuilder = new NodeDiffImpl.Builder<Node>(null, oldPublicQual.get());
+            qualifierDiffBuilder
+                    .withVersionImpact(SemverImpact.MAJOR)
+                    .withMessage("'public' qualifier is removed from class '" + getClassName() + "'")
+                    .build()
+                    .ifPresent(qualifierDiffs::add);
+        }
+
+        NodeList<Token> newQualifiers = newNode.classTypeQualifiers();
+        NodeList<Token> oldQualifiers = oldNode.classTypeQualifiers();
+
+        // analyzes isolated qualifier changes
+        Optional<Token> newIsolatedQual = lookupQualifier(newQualifiers, ISOLATED_KEYWORD);
+        Optional<Token> oldIsolatedQual = lookupQualifier(oldQualifiers, ISOLATED_KEYWORD);
+        if (newIsolatedQual.isPresent() && oldIsolatedQual.isEmpty()) {
+            NodeDiffBuilder qualifierDiffBuilder = new NodeDiffImpl.Builder<Node>(newIsolatedQual.get(), null);
+            qualifierDiffBuilder
+                    .withVersionImpact(SemverImpact.AMBIGUOUS) // TODO: determine compatibility
```

Comment:
Could you plz explian `determine compatibility` mean?

Reply:
currently the version impact of making a non-isolated class an isolated one will be shown as `ambiguous` in the tool, since we can't be sure that its a breaking change or not. Anyway I just added a todo item to check with the language team later and decide whether we can assume about the compatibility more precisely.